src/semanticlayertools/clustering/leiden.py

```
@@ -0,0 +1,174 @@
+    describes the influcence of yearly order on the clustering process. See doc
+    for the Leiden algorithm for more detailed info.
+
+    :param inpath: Path for input network data
+    :type inpath: str
+    :param outpath: Path for writing output data
+    :type outpath: str
+    :param resolution: Main parameter for the clustering quality function (Constant Pots Model)
+    :type resolution: float
+    :param intersliceCoupling: Coupling parameter between two year slices, also influences cluster detection
+    :type intersliceCoupling: float
+    :param timerange: The time range for considering input data (default=1945,2005))
+    :type timerange: tuple
+    :raises OSError: If the output file already exists at class instantiation
+
+    .. seealso::
+       Traag, V.A., Waltman. L., Van Eck, N.-J. (2018).
+       From Louvain to Leiden: guaranteeing well-connected communities.
+       Scientific reports, 9(1), 5233. 10.1038/s41598-019-41695-z
+    """
+
+    def __init__(
+        self, inpath: str, outpath: str,
+        resolution: float = 0.003,
+        intersliceCoupling: float = 0.4,
+        timerange: tuple = (1945, 2005),
+        useGC: bool = True,
+    ):
+        starttime = time.time()
+        self.inpath = inpath
+        self.outpath = outpath
+        self.res_param = resolution
+        self.interslice_param = intersliceCoupling
+        self.timerange = timerange
+
+        self.outfile = os.path.join(
+            outpath,
+            f'timeclusters_{timerange[0]}-{timerange[1]}_res_{resolution}_intersl_{intersliceCoupling}.csv'
+        )
+        if os.path.isfile(self.outfile):
+            raise OSError(f'Output file at {self.outfile} exists. Aborting.')
+
+        if useGC is True:
+            edgefiles = [x for x in os.listdir(inpath) if x.endswith('_GC.net')]
+        elif useGC is False:
+            edgefiles = [x for x in os.listdir(inpath) if x.endswith('.ncol')]
```

If you check for `True` and `False`, it means there's a third option. I think you should raise an exception in that case, or just use `else` instead of `elif`.